### PR TITLE
feat(issueless events) Skip group specific phases in post_process for issueless events

### PR DESCRIPTION
**Pull request by fpacifici**
We cannot skip all post process for issueless events. Event_process signal should be send anyway.
If the event has no group this skips:
- group assignment
- the whole rule processor phase
- service hooks (for now at least)
- plugins

### CODE REVIEW

**Comment by tkaemming** [comment]
Can't dismiss my review as a code owner (for [similarity](https://github.com/getsentry/sentry/blob/78b51778f1f9671bdea480b08781a174ec81fe91/.github/CODEOWNERS#L17)) apparently.

The similarity stuff looks fine, since it's guarded by this check (looks like that comment made it into `master` from #13822 after all): https://github.com/getsentry/sentry/blob/78b51778f1f9671bdea480b08781a174ec81fe91/src/sentry/similarity/features.py#L105-L107

**Comment by fpacifici** [comment]
oops, I forgot about that. removed here.

File: src/sentry/tasks/post_process.py
Comment (fpacifici) [inline src/sentry/tasks/post_process.py]: This is the only statement we need to run out of post_processing for issueless events.
When/if this becomes a performance problem. This can be moved earlier (directly into eventstream) and skip the post process task altogether. That will be a simple change

File: src/sentry/tasks/post_process.py
Comment (tkaemming) [inline src/sentry/tasks/post_process.py]: Shouldn't this also be contained within the `event.group_id` conditional? It doesn't seem like `process_snoozes` expects a nullable group.

**Review by tkaemming — CHANGES_REQUESTED** [review]
One minor issue inline, everything else looks good.

File: src/sentry/tasks/post_process.py
Comment (fpacifici) [reply src/sentry/tasks/post_process.py]: thanks for noticing. Yes it does, this is rebase mistake. Will fix

**Comment by fpacifici** [comment]
Split the PR to make it less scary to review. This one only updates event_processed signal to avoid the dependency on event and logs cases where group information is not consistent with event information (project_id and platform). So if there is no inconsistency we can fully drop the dependency on groups.

**Tests performed:**
look through the codebase to find funcitons conencting to event_processed signal
to ensure all of them do not depend on the group anymore:

```
git grep -A 1 event_processed.connect
src/sentry/receivers/features.py:@event_processed.connect(weak=False)
src/sentry/receivers/features.py-def record_event_processed(project, event, **kwargs):
--
src/sentry/receivers/onboarding.py:@event_processed.connect(weak=False)
src/sentry/receivers/onboarding.py-def record_release_received(project, event, **kwargs):
--
src/sentry/receivers/onboarding.py:@event_processed.connect(weak=False)
src/sentry/receivers/onboarding.py-def record_user_context_received(project, event, **kwargs):
--
src/sentry/receivers/onboarding.py:@event_processed.connect(weak=False)
src/sentry/receivers/onboarding.py-def record_sourcemaps_received(project, event, **kwargs):
--
src/sentry/receivers/similarity.py:@event_processed.connect(weak=False)
src/sentry/receivers/similarity.py-def record(project, event, **kwargs):
```

All addressed

Unit tests for features.py onboarding.py updated accordingly

Logged the execution of all the functions connected to the event_processed signal
```
22:47:22 worker  | FEATURES END python
22:47:22 worker  | RELEASE END
22:47:22 worker  | USER END
22:47:22 worker  | SOURCE START
22:47:22 worker  | SIMILARITY START
```

File: src/sentry/tasks/post_process.py
Comment (tkaemming) [inline src/sentry/tasks/post_process.py]: 😰

**Review by tkaemming — APPROVED** [review]

File: src/sentry/tasks/post_process.py
Comment (fpacifici) [reply src/sentry/tasks/post_process.py]: let's hope it does not log anything. If so, we can rely on events instead of groups and remove this code

**Pull request merged**
